### ISSUE TITLE
[Question] wait for element value text to be NOT empty

### ISSUE THREAD

**Issue by tiger360** [microsoft/playwright#14592]
Hello 😎,
Is there a way to wait until the element value is not empty ,
I tried with "waitForFunction " but I had an error :  page.waitForFunction: call to eval() blocked by CSP
It's on youtube and using Firefox on Ubuntu 20.04.
```js
const selector = 'a.style-scope.ytcp-video-info';
await page.waitForFunction(selector => document.querySelector(selector).getAttribute("href"), selector);
```
Thanks

**Comment by aslushnikov** [comment]
@tiger360 you should use [web-first assertions](https://playwright.dev/docs/test-assertions) for this:

```js
await expect(page.locator('a.style-scope.ytcp-video-info')).toHaveAttribute('href', /.+/);
```

**Comment by tiger360** [comment]
oooh awesome thank you very much, I did not know about it 😉
Is it mandatory to use it with : test('example test', async ({ page }) => { }
because I have this error message : 
`Error: toHaveAttribute must be called during the test`
My code is running in a regular js file.

**Comment by aslushnikov** [comment]
> Is it mandatory to use it with : test('example test', async ({ page }) => { }

@tiger360 Yeah, that's a testing feature, so should be inside a test.

**Comment by pavelfeldman** [comment]
Closing as per above, please feel free to open a new issue if this does not cover your use case.
